rfcs/0169-feature-parameter-names.md

```diff
@@ -0,0 +1,529 @@
+---
+feature: feature_parameter_names
+start-date: 2024-01-10
+author: Dmitry Bogatov
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Nixpkgs has mechanisms to exports compile-time options of the upstream build
+system through the named parameters of derivation function, but the names of
+these parameters are inconsistent from package to package and from feature to
+feature.
+
+# Motivation
+[motivation]: #motivation
+
+Developer interface for building environment with specific features enabled and
+disabled is more complicated and requires more knowledge of implementation
+details compared to already existing systems.
+
+For example, Gentoo Linux has exhaustive list of [all known configuration
+flags](https://www.gentoo.org/support/use-flags) and has means to
+programmatically query what flags are available for particular package. Nixpkgs
+has neither, and this RFC strives to rectify it.
+
+# Detailed design
+[design]: #detailed-design
+
+When upstream build system has options to enable or disable a optional features
+or configure some compile-time parameter, like path to default mailbox or
+initial size of hash map, derivation function (for short, package) MAY expose
+compile-time options via feature parameters that match
+`^(with|conf|enable)_[a-z_0-9]$` regular expression. For short, we will refer
```

Comment:
I'm open to using underscores in these flags, but I think it would require changes to https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax, which says

> Use `lowerCamelCase` for variable names, not `UpperCamelCase`. Note, this rule does not apply to package attribute names, which instead follow the rules in [package naming](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming).

Reply:
Thank you for bringing it up. I'll add into the migration section step of updating this documentation.